end_to_end_tests/test_custom_templates/endpoint_init.py.jinja

```diff
@@ -0,0 +1,37 @@
+""" Contains methods for accessing the API Endpoints """
+
+{% macro snake_to_pascal_case(name) %}
+{%- for part in name.split('_') -%}
+{{ part | capitalize }}
+{%- endfor -%}
+{% endmacro %}
+
+{% macro to_snake_name(name) %}
+{% set last_snakize = {'at': 0} %}
+{%- for part in name -%}
+{{ '_' if not loop.first and part in 'ABCDEFGHIJKLMNOPQRSTUVWXYZ' and (loop.index - last_snakize.at) > 1 and not last_snakize.update({'at': loop.index}) }}{{ part | lower }}
+{%- endfor -%}
+{% endmacro %}
```

Comment:
this macro will render wrong names `module__enpoint_func` with extra underscore.
@dbanty whouldn't it be nice to expose utils methods to jinja templates?

by just adding to Project class
```
        self.env.globals.update(snake_case=snake_case, pascal_case=pascal_case)
```

Reply:
Probably? I don't use custom templates at all and they aren't considered stable, I just mostly approve other folks' contributions for now so they can do what they need. If having those available in custom templates would be useful to you then we can totally pass them through. I think in the primary templates though we don't do any string transformations, it's all done in the Python parsing code.

Reply:
I agree, it will be nice to have, best I thinks will be to make  https://github.com/triaxtec/openapi-python-client/blob/main/openapi_python_client/utils.py accessible to jinja.

Reply:
I've introduce the change in https://github.com/triaxtec/openapi-python-client/pull/403/commits/8edfaa6c7a176273f1b5b40a903b3b169282c902 and updated the custom template sample in https://github.com/triaxtec/openapi-python-client/pull/403/commits/9907434a6172a5c211325a95db08c6ead7228ab1